synapse/http/proxyagent.py

```diff
@@ -243,28 +244,45 @@ def request(self, method, uri, headers=None, bodyProducer=None):
         )
 
 
-def _http_proxy_endpoint(proxy: Optional[bytes], reactor, **kwargs):
+def _http_proxy_endpoint(
+    proxy: Optional[bytes],
+    reactor: IReactorCore,
+    tls_options_factory: Optional[IPolicyForHTTPS],
+    **kwargs,
+):
     """Parses an http proxy setting and returns an endpoint for the proxy
 
     Args:
-        proxy: the proxy setting in the form: [<username>:<password>@]<host>[:<port>]
-            Note that compared to other apps, this function currently lacks support
-            for specifying a protocol schema (i.e. protocol://...).
+        proxy: the proxy setting in the form: [<username>:<password>@][scheme://]<host>[:<port>]
```

Comment:
This comment isn't exactly correct (this methods doesn't handle the `username:password` stuff, that's done before) but this was here before, so I went with not touching it here.

Reply:
If you're changing it, why not fix it.

It's hard to see that putting the username and password *before* the scheme could possibly make any sense.

Reply:
> It's hard to see that putting the username and password before the scheme could possibly make any sense.

Ugh, that wasn't really the intention here... :-/

/me goes and looks if only the comment is wrong or the whole thing :(